components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -1836,73 +1836,93 @@ public IdentityProvider getEnabledIdPByRealmId(String realmId, String tenantDoma
-                                                    spName = properties[b].getValue();
+                                            for (Property value : properties) {
+                                                if (value != null && value.getName() != null &&
+                                                        value.getName()
+                                                                .equals(IdentityApplicationConstants.Authenticator.SAML2SSO.SP_ENTITY_ID)) {
+                                                    spName = value.getValue();
                                                 }
                                             }
                                             if (spName.equals("")) {
                                                 throw new IdentityProviderManagementException("SP name can't be empty");
                                             }
 
-                                            if (metaFederated != null && ArrayUtils.isNotEmpty(metaFederated.getProperties())) {
+                                            if (metaFederated != null &&
+                                                    ArrayUtils.isNotEmpty(metaFederated.getProperties())) {
                                                 for (int y = 0; y < metaFederated.getProperties().length; y++) {
-                                                    if (metaFederated.getProperties()[y] != null && metaFederated.getProperties()[y].getName() != null
-                                                            && metaFederated.getProperties()[y].getName().toString().equals(IdentityApplicationConstants.Authenticator.SAML2SSO.SP_ENTITY_ID)) {
+                                                    if (metaFederated.getProperties()[y] != null &&
+                                                            metaFederated.getProperties()[y].getName() != null
+                                                            && metaFederated.getProperties()[y].getName()
+                                                            .equals(IdentityApplicationConstants.Authenticator.SAML2SSO.SP_ENTITY_ID)) {
                                                         metaFederated.getProperties()[y].setValue(spName);
                                                         break;
                                                     }
                                                 }
                                             }
 
-                                            if (metaFederated != null && metaFederated.getProperties() != null && metaFederated.getProperties().length > 0) {
-                                                federatedAuthenticatorConfigs[i].setProperties(metaFederated.getProperties());
+                                            if (metaFederated != null && metaFederated.getProperties() != null &&
+                                                    metaFederated.getProperties().length > 0) {
```

Comment:
Can we use ArrayUtils here so that we can check for both null and emptiness at the same time?